src/main/java/com/kodilla/ecommercee/domain/cart/CartEntity.java

```diff
@@ -39,7 +41,10 @@ public void setCost(BigDecimal cost) {
         this.cost = cost;
     }
 
-    @ManyToMany(cascade = CascadeType.ALL)
+    @ManyToMany(cascade = {
+            CascadeType.PERSIST,
+            CascadeType.MERGE
+    })
```

Comment:
nie wiem czy nie warto dodać też CascadeType.REFRESH w raie gdyby coś w encji User się zmieniło np. ID, ale to luźna propozycja

Reply:
> Waro by jeszcze posprawdzać w teście Id usera i produktów w asercji.

Wydaje mi się że testy klasy Cart, nie powinny sprawdzać czy użytkownik jest poprawnie zapisywany, bo powinny to robić testy klasy User.
A co do produktów to 
`        Assert.assertNotEquals(Optional.of(0L), readProduct1.get().getId());
         Assert.assertNotEquals(Optional.of(0L), readProduct2.get().getId());
         Assert.assertNotEquals(Optional.of(0L), readProduct3.get().getId());`
tutaj sprawdzam czy produkty są w ogóle zapisywane, nie mogę wstawić na sztywno liczby, bo są generowane
![image](https://user-images.githubusercontent.com/34661411/71787320-8fe84880-3016-11ea-93ae-d9b4c4b9723a.png)
![image](https://user-images.githubusercontent.com/34661411/71787328-9d053780-3016-11ea-9bc9-3dc2fb0469e8.png)
![image](https://user-images.githubusercontent.com/34661411/71787335-b0180780-3016-11ea-8395-59257fc1ca5f.png)

> nie wiem czy nie warto dodać też CascadeType.REFRESH

Co do tego nie jestem pewny, może niech chłopaki też się wypowiedzą

Reply:
Ok u mnie testy dla encji user z cart działają, powinniśmy mimo wszystko sprawdzać wszystkie relacje w każdą stronę wiec jak mamy cart to ma on relacje z product order i user wiec wypada sprawdzić wszystkich poprawność. W bazie z testami H2 jest prościej z tymi Id bo za każdym razem startuje od początku i generuje kolejno nie ważne czy cleanup robimy czy nie.